### DISCUSSION TITLE
ffmpeg_avcodec_log bad cseq

### DISCUSSION THREAD

**Discussion by angrycoding** [Motion-Project/motion#1485] [General]
Hi, I'm getting this in motion.log

[2:nc2:00:12:31:04] [INF] [ENC] [May 16 22:19:20] ffmpeg_avcodec_log: RTP: PT=60: bad cseq 0000 expected=d394
[2:nc2:00:12:31:04] [INF] [ENC] [May 16 22:19:20] ffmpeg_avcodec_log: RTP: PT=60: bad cseq d394 expected=000a
[2:nc2:00:12:31:04] [INF] [ENC] [May 16 22:19:20] ffmpeg_avcodec_log: RTP: PT=60: bad cseq 000a expected=d39e
[2:nc2:00:12:31:04] [INF] [ENC] [May 16 22:19:20] ffmpeg_avcodec_log: RTP: PT=60: bad cseq d39e expected=0014
[2:nc2:00:12:31:04] [INF] [ENC] [May 16 22:19:20] ffmpeg_avcodec_log: RTP: PT=60: bad cseq 0014 expected=d3a8
[2:nc2:00:12:31:04] [INF] [ENC] [May 16 22:19:20] ffmpeg_avcodec_log: RTP: PT=60: bad cseq d3a8 expected=001e

I understand that the issue is related with the corrupted RTSP stream, but there is no reaction from motion side on it, which means that motion is continuing to detect motion events in a corrupted stream which leads to constantly emiting video fragments.

Would be great if you can fix it in one of the following ways:

- in case if this happens, motion could emit some event (on_stream_error for instance), so backend can react on it and reboot the camera for instance
- allow somehow to set format for logging messages so camera ID will be printed into log message, this can also be nice to be done by default (without any formatting specifiers). In this case backend can constantly check log to see if there are any stream - related issues and reboot camera which issues the corrupted stream for instance.

**Comment by tosiara** [comment]
What issue do you have? Corrupted picture? Can you please elaborate

**Comment by angrycoding** [comment]
Well it's not that easy to explain. So I've a camera connected to motion in a regular way. One highres stream for recording and another lores for detection. Sometimes weird things starting to happen. First of all I see tones of such messages coming into log file (bad cseq). In the mean time i see that motion starts to detect motion on video artefacts produced by corrupted stream. That at the end leads to issuing dozens of motion events with no actual movement. There are several problems:
- motion does not monitor (or not doing it properly) corrupted stream reported by ffmpeg. 
- does not provide the end user with some mechanism to determine such corruption. Would be nice to have on_stream_error event or something.
- does not print camera id into the log file, so it's impossible to determine which camera produced such corrupted stream.

So in order to improve this situation, would be nice to have:
- when motion detects stream corruption (just inspect stderr or out from ffmpeg), it triggers on_stream_error event or such, so external script can handle this situation (send reboot to the camera, or disable it)
- when issuing log messages into log file, would be nice to print camera id (as its defined in canera config)

**Comment by tosiara** [comment]
Are you using RTSP over UDP?

**Comment by angrycoding** [comment]
Nope, tcp only. It doesn't happen often, for instance last no errors period was about two weeks long. Then at some point, baaang, and Im getting it. Now the only way for me to fix that, is rebooting camera manually after tailing log file. If you need log, i can give it to you.

And one more time: i dont think that its motion or ffmpeg issue. My camera is one of those cheapest and crappy AliExpress cams, so i blame camera. Whole this thing is more about letting user know that something bad is going on, rather than investigating ffmpeg codec streams tcp udp stuff (that might also be good of course, but I think its camera issue at the end).

**Comment by tosiara** [comment]
There were few discussions about corrupted RTSP frames:
https://sourceforge.net/p/motion/mailman/message/32355783/
https://github.com/Motion-Project/motion/issues/503
https://sourceforge.net/p/motion/mailman/message/36529258/

So far there is no way for motion to know whether frame was corrupted or not. Decoder either provides a decoded picture as is or fails. Both ffmpeg and libjpeg can throw warnings, but that does not affect the process

I have multiple cheap cameras that produce tons of ffmpeg errors, but work fine. So despite the that warning motion does not know if frame was fully decoded or just partially

**Comment by angrycoding** [comment]
Okay, but what about logging camera id into log file? So far that would be acceptable solution for monitoring such problems.

**Comment by tosiara** [comment]
My log looks like this:
```
May 30 22:33:40 mk808b motion: [1:ml1:mk808b] [NTC] [EVT] event_create_extpipe: cnt->moviefps: 8
May 30 22:33:40 mk808b motion: [1:ml1:mk808b] [NTC] [EVT] event_newfile: File of type 8 saved to: /home/motion/cam0/6807-20190530_22-33
May 30 22:33:40 mk808b motion: [1:ml1:mk808b] [NTC] [ALL] motion_detected: Motion detected - starting event 6807
```
Here mk808b is the camera id

In your log I see `[2:nc2:00:12:31:04] ` which needs some check, I have no idea so far what this output means as well as I haven't seen your full log

**Comment by angrycoding** [comment]
Haha, yeah, now its starting to make sense, the thing is that I use mac addresses as camera name, so it looks like motion just takes a part of it and the rest is removed. 
So the only issue is that wrong size stated in the documentation (Range / Valid values: Max 4095 characters), or there is a bug in logger, which silently cut camera_name in case if its exceeds 11 characters.

**Comment by angrycoding** [comment]
This is how my config look like:

camera_name 00:12:31:04:10:56
netcam_url rtsp://127.0.0.1:554//user=admin_password=lXlFDZPY_channel=1_stream=1.sdp
netcam_highres rtsp://127.0.0.1:554//user=admin_password=lXlFDZPY_channel=1_stream=0.sdp
width 704
height 576
movie_passthrough on

**Comment by tosiara** [comment]
The log prints the thread name which is built from internal thread name and camera name variable and is hardcoded to 16 chars: https://github.com/Motion-Project/motion/blob/master/motion.c#L4093
You can also try to define `camera_id` if you like
Check the documentation: https://motion-project.github.io/motion_config.html#camera_id
